stats_test.go

```diff
@@ -208,3 +234,187 @@ func TestNilAggregation(t *testing.T) {
 		t.Errorf("Expected: %v, Got: %v", fmt.Errorf("aggregation *view.Aggregation is not supported"), actual)
 	}
 }
+
+func TestPercentileName(t *testing.T) {
+	testCases := []struct {
+		Percentile float64
+		Expected   string
+	}{
+		{
+			0.5,
+			"50percentile",
+		},
+		{
+			0.75,
+			"75percentile",
+		},
+		{
+			0.92,
+			"92percentile",
+		},
+		{
+			0.95,
+			"95percentile",
+		},
+		{
+			0.99,
+			"99percentile",
+		},
+		{
+			0.995,
+			"995percentile",
+		},
+		{
+			0.999,
+			"999percentile",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%f", tc.Percentile), func(t *testing.T) {
+			got := percentileName(tc.Percentile)
+			if got != tc.Expected {
+				t.Errorf("Expected: %v, Got %v\n", tc.Expected, got)
+			}
+		})
+	}
+}
+
+func TestCalculatePercentile(t *testing.T) {
+	var buckets []float64
+	for i := 0.01; i < 100; i += 0.01 {
+		buckets = append(buckets, i)
+	}
+
+	var countsPerBucket []int64
+	for i := int64(0); i <= int64(len(buckets)); i += 1 {
+		countsPerBucket = append(countsPerBucket, 10)
+	}
+
+	testCases := []struct {
+		expected        float64
+		percentile      float64
+		buckets         []float64
+		countsPerBucket []int64
+	}{
+		{
+			0,
+			0,
+			buckets,
+			countsPerBucket,
+		},
+		{
+			10,
+			0.1,
+			buckets,
+			countsPerBucket,
+		},
+		{
+			50,
+			0.5,
+			buckets,
+			countsPerBucket,
+		},
+		{
+			70,
+			0.70,
+			buckets,
+			countsPerBucket,
+		},
+		{
+			81,
+			0.81,
+			buckets,
+			countsPerBucket,
+		},
+		{
+			100,
+			1.0,
+			buckets,
+			countsPerBucket,
+		},
+		{
+			99,
+			0.99,
+			buckets,
+			countsPerBucket,
+		},
+		{
+			99.9,
+			0.999,
+			buckets,
+			countsPerBucket,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%v", tc.percentile), func(t *testing.T) {
+			got := calculatePercentile(tc.percentile, tc.buckets, tc.countsPerBucket)
+
+			if math.Abs(tc.expected-got) > 0.01 {
+				t.Errorf("Expected: %v to be within 0.1 of %v", tc.expected, got)
+			}
+		})
+
+	}
+}
+
+func TestCalculatePercentileTwoPower(t *testing.T) {
+	buckets := []float64{1, 2, 4, 8, 16, 32, 64, 128, 256}
+
```

Comment:
I think you can remove this empty line, and the one below too.

Reply:
Sure. Fixed.